distributed/joblib.py

```diff
@@ -35,5 +35,11 @@ def callback_wrapper():
         future.get = future.result  # monkey patch to achieve AsyncResult API
         return future
 
+    def abort_everything(self, ensure_ready=True):
+        # Tell the executor to cancel any task submitted via this instance
+        # as joblib.Parallel will never access those results.
+        executor = self.executor
+        executor.cancel([Future(k, executor) for k in executor.futures])
```

Comment:
@mrocklin the current cancel method does not accept future keys as argument hence the need to manually wrap as a Future instance. I wonder if the `Executor.cancel` API should be changed to make it possible not have to create those intermediate objects. Not a big deal though.